### PR TITLE
Add support for Python 3.5

### PR DESCRIPTION
**Pull request by dmtucker**
This is needed for (and pretty much identical to) https://github.com/henry0312/pytest-docstyle/pull/18

### CODE REVIEW

**Comment by dmtucker** [comment]
I think a better solution than manually adding defaults to the help text would be to use [`argparse.ArgumentDefaultsHelpFormatter`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter), but I'm not sure if that's possible with Pytest.

File: setup.py [hunk 1]
```diff
@@ -17,16 +17,14 @@
...
         install_requires=[
             'pytest>=3.0,<4',
             'py>=1.5,<1.6',
             'pycodestyle>=2.3,<2.4',
         ],
         extras_require={
-            'tests': [
-                'pytest-isort>=0.1,<1.0',
-            ],
+            'tests': ['tox'],
```
Comment (henry0312) [inline setup.py]: I don't think tox is needed.

File: pytest_codestyle.py [hunk 2]
```diff
@@ -4,19 +4,18 @@
...
     parser.addini('codestyle_max_line_length', default=pycodestyle.MAX_LINE_LENGTH,
-                  help=f"set maximum allowed line length (default: {default_max_line_length})")
+                  help='set maximum allowed line length (default: {max_line_length})'.format(
+                      max_line_length=pycodestyle.MAX_LINE_LENGTH))
     parser.addini('codestyle_select', type='args',
                   help='select errors and warnings (default: [])')
     parser.addini('codestyle_ignore', type='args',
-                  help=f"skip errors and warnings (default: [{' '.join(default_ignore)}])")
+                  help='skip errors and warnings (default: [{ignored}])'.format(
+                      ignored=pycodestyle.DEFAULT_IGNORE.replace(',', ' ')))
```
Comment (henry0312) [inline pytest_codestyle.py]: 🆒

**Comment by henry0312** [comment]
Unfortunately, the tests with Python 3.5 fails because of [PEP 519: Adding a file system path protocol](https://docs.python.org/3/whatsnew/3.6.html#pep-519-adding-a-file-system-path-protocol)

cf. [python - When using pathlib, getting error: TypeError: invalid file: PosixPath('example.txt') - Stack Overflow](https://stackoverflow.com/questions/42694112/when-using-pathlib-getting-error-typeerror-invalid-file-posixpathexample-t/42694113#42694113)

cf. test log https://gist.github.com/henry0312/3f2fbe4221a8c573f453105e1063ab36

I'm sorrty I tell you that I can't merge this PR until the problem is solved.

**Comment by dmtucker** [comment]
Ah, took me a while to figure that one out... Pytest uses `pathlib` and claims support for Python 3.5 (fine as long as they don't pass it to `open`), and Pycodestyle calls `open` and claims support for 3.5 (fine as long as we pass a string instead of whatever `pytest` gives us).

File: pytest_codestyle.py [hunk 3]
```diff
@@ -44,7 +43,7 @@ def runtest(self):
         # http://pycodestyle.pycqa.org/en/latest/api.html#pycodestyle.Checker
         # http://pycodestyle.pycqa.org/en/latest/advanced.html
         checker = pycodestyle.Checker(
-                        filename=self.fspath,
+                        filename=str(self.fspath),
```
Comment (henry0312) [inline pytest_codestyle.py]: 👍

**Comment by henry0312** [comment]
Thank you!

**Comment by henry0312** [comment]
I'll release a new version soon.

**Pull request merged**